### ISSUE TITLE
Removing a Node doesn't work

### ISSUE THREAD

**Issue by yvespp** [yamamoto-febc/terraform-provider-rke#48]
I built an rke cluster on fresh VMs via:
```
terraform apply -var node_ips='["10.22.141.17","10.22.141.18","10.22.141.19"]'
```
TF file:
```
variable "node_ips" {
  type = "list"
  description = "List of node ips to provision"
}

locals {
  count = "${length(var.node_ips)}"
}

data rke_node_parameter "nodes" {
  count   = "${local.count}"
  address = "${var.node_ips[count.index]}"
  user    = "rancher"
  role    = ["controlplane", "worker", "etcd"]
}

resource rke_cluster "cluster" {
  nodes_conf = ["${data.rke_node_parameter.nodes.*.json}"]
  ssh_agent_auth = true
  ....omitted....
}
resource "local_file" "kube_cluster_yaml" {
  filename = "${path.root}/kube_config_cluster.yml"
  content = "${rke_cluster.cluster.kube_config_yaml}"
}

resource "local_file" "rke_cluster_yaml" {
  filename = "${path.root}/cluster.yml"
  content = "${rke_cluster.cluster.rke_cluster_yaml}"
}
```
As expected a cluster with 3 Nodes gets created.
I then remove an ip from the `node_ips` var an rerun terraform:
```
$ terraform apply -var node_ips='["10.22.141.17","10.22.141.18"]'
data.rke_node_parameter.nodes[0]: Refreshing state...
data.rke_node_parameter.nodes[1]: Refreshing state...
rke_cluster.cluster: Refreshing state... (ID: 10.22.141.17)
local_file.kube_cluster_yaml: Refreshing state... (ID: f7c055d88ecb4f086055cf9bcd6df0a10c1faaff)
local_file.rke_cluster_yaml: Refreshing state... (ID: 15cd80558fb10b085fb7f0d636f6e490b26b80b2)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.rke.rke_cluster.cluster
      nodes_conf.#: "3" => "2"
      nodes_conf.2: "{\"address\":\"10.22.141.19\",\"role\":[\"controlplane\",\"worker\",\"etcd\"],\"user\":\"rancher\"}" => ""


Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.rke.rke_cluster.cluster: Modifying... (ID: 10.22.141.17)
  nodes_conf.#: "3" => "2"
  nodes_conf.2: "{\"address\":\"10.22.141.19\",\"role\":[\"controlplane\",\"worker\",\"etcd\"],\"user\":\"rancher\"}" => ""
module.rke.rke_cluster.cluster: Still modifying... (ID: 10.22.141.17, 10s elapsed)
module.rke.rke_cluster.cluster: Still modifying... (ID: 10.22.141.17, 20s elapsed)
module.rke.rke_cluster.cluster: Still modifying... (ID: 10.22.141.17, 30s elapsed)
module.rke.rke_cluster.cluster: Still modifying... (ID: 10.22.141.17, 40s elapsed)
module.rke.rke_cluster.cluster: Still modifying... (ID: 10.22.141.17, 50s elapsed)
module.rke.rke_cluster.cluster: Modifications complete after 58s (ID: 10.22.141.17)

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```
However `kubectl` still shows 3 nodes:
```
$ kubectl --kubeconfig=kube_config_cluster.yml get nodes
NAME           STATUS   ROLES                      AGE     VERSION
10.22.141.17   Ready    controlplane,etcd,worker   8m31s   v1.12.4
10.22.141.18   Ready    controlplane,etcd,worker   8m31s   v1.12.4
10.22.141.19   Ready    controlplane,etcd,worker   8m31s   v1.12.4
```
Also the `cluster.yml` still contains the `10.22.141.19` Node. Only after running `terraform apply` again the `cluster.yml` is updated. It seems the `cluster.yml` is created before the cluster is updated.

Running `rke up` by hand with the generated `cluster.yml` (`10.22.141.19` is missing in the file) also doesn't remove the Node.
Building the 3 Node cluster with terraform, removing a Node from the `cluster.yml` by hand and running `rke up` removes the Node correctly from the cluster.

Versions:
```
$ terraform version
Terraform v0.11.11
+ provider.local v1.1.0
+ provider.rke v0.7.0
```

**Comment by yamamoto-febc** [comment]
Hi @yvespp, 

Thank you for your report and detailed explanation!

It seems there are two problems as follows:
- Nodes aren't removed when `terraform apply` running.
- `rke_cluster_yaml` is not changed when node defination is changed on tf file.

First, I investigated RKE and noticed that RKE uses generated `kube_config_cluster.yml` when updating cluster.
https://github.com/rancher/rke/blob/v0.1.15/cluster/cluster.go#L169-L172

If `kube_config_cluster.yml` is not found when updating cluster, RKE doesn't reconcile cluster state.
https://github.com/rancher/rke/blob/v0.1.15/cluster/reconcile.go#L26-L30

However `terraform-provider-rke` doesn't prepare `kube_config_cluster.yml` before performing cluster update.

I think this is the cause of the first problem.
So I'm going to fix this.

Next, `rke_cluster_yaml` is currently set only after `clusterUp` is called.
But I think `rke_cluster_yaml` can set before `clusterUP` is called by reading information from tf files.
I'm going to fix this too.

**Comment by yamamoto-febc** [comment]
I just released [v0.7.1](https://github.com/yamamoto-febc/terraform-provider-rke/releases/tag/0.7.1).

This version includes some fixes. Please try it:pray:

**Comment by mkozjak** [comment]
@yamamoto-febc: For me docker build fails with:
```
go: verifying github.com/hashicorp/go-getter@v0.0.0-20180809191950-4bda8fa99001: checksum mismatch
	downloaded: h1:qC+3MHkvfCXb1cA9YDpWZ7np8tPOXZceLrW+xyqOgmk=
	go.sum:     h1:MFPzqpPED05pFyGjNPJEC2sXM6EHTzFyvX+0s0JoZ48=
make: *** [build] Error 1
```

Got this basically from trying to build from latest commit with `docker build --tag rke .`.

**Comment by yamamoto-febc** [comment]
@mkozjak Thank you for reporting!

I created new issue #56.

**Comment by yvespp** [comment]
I tested it and both issues are resolved: nodes are correctly removed and the `cluster.yml` is updated correctly as well.
Thanks a lot!

**Comment by mkozjak** [comment]
Awesome work! Thanks, guys!
